### PR TITLE
apps-sc: Updated harbor chart to work in HA

### LINKED ISSUE #854: [1] Stability: Make Harbor HA
**What should be investigated.**

Please investigate what is needed to make Harbor highly available.
Any single Node failure should not impact metrics collection.

**Implementation hints**

* https://goharbor.io/docs/1.10/install-config/harbor-ha-helm/
* Our managed Redis and PostgreSQL operators

**What artifacts should this produce.**

* A demo showing that this works, i.e., Nodes hosting Harbor components can be failed without impacting pushing/pulling container images.
* An outline of changes needed in Compliant Kubernetes.
* An outline of migration steps needed to be performed.

Depends on: https://github.com/elastisys/compliantkubernetes-postgresql/issues/111 and https://github.com/elastisys/compliantkubernetes-redis/issues/54

### PR DESCRIPTION
**Pull request by Xartos**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #854

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/values/harbor.yaml.gotmpl [hunk 1]
```diff
@@ -110,6 +117,10 @@ registry:
   tolerations:  {{- toYaml .Values.harbor.tolerations | nindent 4 }}
 
 notary:
```
Comment (crssnd) [inline helmfile/values/harbor.yaml.gotmpl:119]: seems to be another notary key starting with line 173.
I didn't manage to find if it's a duplicate or not

Comment (Xartos) [reply helmfile/values/harbor.yaml.gotmpl:119]: Oooh, nice find! Seems to be a duplicate. Let me merge those two

**Review by Pavan-Gunda — APPROVED** [review]
LGTM!!

File: migration/v0.24.x-v0.25.x/migrate-harbor-affinity-variables.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (crssnd) [inline migration/v0.24.x-v0.25.x/migrate-harbor-affinity-variables.sh:1]: please make the script executable

File: config/config/sc-config.yaml [hunk 3]
```diff
@@ -192,6 +261,20 @@ harbor:
         requests:
           memory: 32Mi
           cpu: 10m
+      affinity:
```
Comment (crssnd) [inline config/config/sc-config.yaml:264]: not sure if necessary as it will always run with 1 replica

File: config/config/sc-config.yaml [hunk 4]
```diff
@@ -137,6 +164,20 @@ harbor:
         requests:
           memory: 512Mi
           cpu: 250m
+      affinity:
```
Comment (crssnd) [inline config/config/sc-config.yaml:167]: not sue if necessary here

File: config/config/sc-config.yaml (hunk 3 above)
Comment (Xartos) [reply config/config/sc-config.yaml:264]: Sure, but it doesn't hurt and then if you scale up to 2 replicas you already have the spreading already in the defaults.

Comment (crssnd) [reply config/config/sc-config.yaml:264]: yes it doesn't hurt, but from what I can see scaling for internal database and redis is not supported by the chart

**Review by crssnd — APPROVED** [review]

**Review by raviranjanelastisys — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]

**Pull request merged**